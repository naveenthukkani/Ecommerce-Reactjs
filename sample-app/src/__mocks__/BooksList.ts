import {Books} from "../../src/home/Books";

export const BooksList: Array<Books> = [
   {
       name:"maths",
       discription: "it is simple problems book for children",
       author:"naveen",
       price: 5,
       pagecount: 1,
	   isbn:"",
	   id:1,
   },
   {
    name:"social",
    discription: "it is sample book for children",
    author:"naveen",
    price: 7,
    pagecount: 1,
	isbn:"",
	id:2,
},
{
    name:"telugu",
    discription: "it is language learning book",
    author:"naveen",
    price: 8,
    pagecount: 1,
	isbn:"",
	id:3,
},
{
    name:"English",
    discription: "it is international language learning book",
    author:"naveen",
    price: 5,
    pagecount: 1,
	isbn:"",
	id:4,
},
{
    name:"maths",
    discription: "it is simple problems book for children",
    author:"naveen",
    price: 5,
    pagecount: 1,
	isbn:"",
	id:5,
},
{
	name:"social",
	discription: "it is sample book for children",
	author:"naveen",
	price: 7,
	pagecount: 1,
	isbn:"",
	id:6,
},
{
	name:"telugu",
	discription: "it is language learning book",
	author:"naveen",
	price: 8,
	pagecount: 1,
	isbn:"",
	id:7,
},
{
	name:"English",
	discription: "it is international language learning book",
	author:"naveen",
	price: 5,
	pagecount: 1,
	isbn:"",
	id:8,
},
{
	name:"social",
	discription: "it is sample book for children",
	author:"naveen",
	price: 7,
	pagecount: 1,
	isbn:"",
	id:9,
},
{
	name:"telugu",
	discription: "it is language learning book",
	author:"naveen",
	price: 8,
	pagecount: 1,
	isbn:"",
	id:10,
},
{
	name:"English",
	discription: "it is international language learning book",
	author:"naveen",
	price: 5,
	pagecount: 1,
	isbn:"",
	id:11,
}
]