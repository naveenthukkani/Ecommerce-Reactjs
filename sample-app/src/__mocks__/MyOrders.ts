import {Orders} from "../../src/myorders/Orders";

export const OrdersList: Array<Orders> = [
        {
            date: new Date().toLocaleString(),
            status: "Delivered",
            isbn: "9781593275846",
            title: "Eloquent JavaScript, Second Edition",
            subtitle: "A Modern Introduction to Programming",
            author: "Marijn Haverbeke",
            pages: 472,
            price: 500,
            id:1
        },
];