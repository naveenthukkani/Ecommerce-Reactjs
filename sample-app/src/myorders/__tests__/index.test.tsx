import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MyOrders} from '../index';
import { OrdersList } from '../../__mocks__/MyOrders';

describe('MyOrders', () =>{
    let props:any;
    beforeEach(()=>{
        props = {
         ordersList: OrdersList,
        }
    });

    it('renders the component correctly', () => {
        const component = shallow(<MyOrders {...props} />);
        expect(toJson(component)).toMatchSnapshot();
      });
});