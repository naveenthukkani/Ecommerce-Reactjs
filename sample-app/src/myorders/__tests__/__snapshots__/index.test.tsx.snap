// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyOrders renders the component correctly 1`] = `
<div
  className="Orders-list"
>
  <div
    key="1"
  >
    <MyOrder
      author="Marijn Haverbeke"
      date="7/19/2020, 12:30:21 PM"
      id={1}
      isbn="9781593275846"
      pages={472}
      price={500}
      status="Delivered"
      subtitle="A Modern Introduction to Programming"
      title="Eloquent JavaScript, Second Edition"
    />
  </div>
</div>
`;
